website/addons/owncloud/views.py

```diff
@@ -0,0 +1,114 @@
+"""Views for the node settings page."""
+# -*- coding: utf-8 -*-
+import httplib as http
+
+from furl import furl
+import requests
+from flask import request
+from modularodm import Q
+from modularodm.storage.base import KeyExistsException
+from framework.auth.decorators import must_be_logged_in
+
+from website.addons.base import generic_views
+from website.oauth.models import ExternalAccount
+from website.project.decorators import (
+    must_have_addon)
+
+import owncloud
+from website.addons.owncloud.model import OwnCloudProvider
+from website.addons.owncloud.serializer import OwnCloudSerializer
+from website.addons.owncloud import settings
+
+SHORT_NAME = 'owncloud'
+FULL_NAME = 'OwnCloud'
+
+owncloud_account_list = generic_views.account_list(
+    SHORT_NAME,
+    OwnCloudSerializer
+)
+
+owncloud_import_auth = generic_views.import_auth(
+    SHORT_NAME,
+    OwnCloudSerializer
+)
+
+owncloud_deauthorize_node = generic_views.deauthorize_node(
+    SHORT_NAME
+)
+
+owncloud_root_folder = generic_views.root_folder(
+    SHORT_NAME
+)
+
+## Config ##
+
+@must_be_logged_in
+def owncloud_add_user_account(auth, **kwargs):
+    """Verifies new external account credentials and adds to user's list"""
+
+    # Ensure that ownCloud uses https
+    host_url = request.json.get('host')
+    host = furl()
+    host.host = host_url.rstrip('/').strip('https://').strip('http://')
```

Comment:
If it works (and a bit of local testing indicates that it does). ¯\\\_(ツ)_/¯ 

Could you add a test or two, passing poorly formatted urls in the json, for this view function in `owncloud/tests/test_views.py:TestAuthViews`, to prevent any regressions? Some amount of frontend validation / helper-text to indicate proper formatting to the user would be nice as well, but isn't necessarily a blocker.